app/controllers/group_assignment_invitations_controller.rb

```diff
@@ -74,63 +68,50 @@ def create_group_assignment_repo(selected_group: group, new_group_title: nil)
     end
   end
 
-  def group_assignment
-    @group_assignment ||= invitation.group_assignment
+  def current_invitation
+    @invitation ||= GroupAssignmentInvitation
+                    .includes(group_assignment: :group_assignment_repos)
+                    .find_by!(key: params[:id])
   end
-  helper_method :group_assignment
 
-  def group_assignment_repo
-    @group_assignment_repo ||= GroupAssignmentRepo.find_by(group_assignment: group_assignment, group: group)
+  def current_submission
+    @current_submission ||= GroupAssignmentRepo.find_by(group_assignment: current_assignment, group: group)
   end
-  helper_method :group_assignment_repo
 
-  def group_params
-    params
-      .require(:group)
-      .permit(:id, :title)
-  end
+  def ensure_submission_repository_exists
+    return not_found unless current_submission
```

Comment:
`not_found` raises an exception, so I'm not sure returning here really does anything, since it'll never actually return.